src/app/permissions/project-survey-detail/project-survey-detail.component.ts

```diff
@@ -0,0 +1,136 @@
+import { Component, Input, OnInit, OnDestroy } from '@angular/core';
+import { Subject } from 'rxjs/Subject';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Router, ActivatedRoute, ActivatedRouteSnapshot } from '@angular/router';
+
+import { StorageService } from 'app/services/storage.service';
+import { SurveyService } from 'app/services/survey.service';
+
+import { Survey } from 'app/models/survey';
+
+@Component({
+  selector: 'app-project-survey-detail',
+  templateUrl: './project-survey-detail.component.html',
+  styleUrls: ['./project-survey-detail.component.scss']
+})
+export class ProjectSurveyDetailComponent implements OnInit, OnDestroy {
+
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+
+  public commentPeriodPublishedStatus: string;
+  public publishAction: string;
+  public projectId: any;
+  public currentProject: any;
+  public surveyLastSaved: any;
+  public loading;
+  public survey: any;
+  public surveyCopy: any;
+  public countArray = [];
+  public commentPeriodDocs = [];
+  public canDeleteSurvey = false;
+  public prettyCommentingMethod: string;
+  public surveySelected: string | Survey;
+
+  constructor(
+    private storageService: StorageService,
+    private surveyService: SurveyService,
+    private snackBar: MatSnackBar,
+    private router: Router,
+    private route: ActivatedRoute,
+    ) {}
+
+  ngOnInit() {
+      this.route.data
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((data: any) => {
+        if (data.survey) {
+          this.survey = data.survey;
+          this.surveyItemCount(this.survey.questions)
+          this.currentProject = this.storageService.state.currentProject.data;
+          this.projectId = this.storageService.state.currentProject.data._id;
+          this.loading = false;
+        } else {
+          alert('Uh-oh, couldn\'t survey. It may no longer exist.');
+          // survey not found --> navigate back to search
+          this.router.navigate(['/search']);
+        }
+      });
+
+  }
+
+  openSnackBar(message: string, action: string) {
+    this.snackBar.open(message, action, {
+      duration: 2000,
+    });
+  }
+
+  editSurvey() {
+    this.router.navigateByUrl(`/p/${this.projectId}/s/${this.survey._id}/edit`);
+  }
+
+  deleteSurvey() {
+    if (confirm(`Are you sure you want to delete this survey?`)) {
+      this.loading = true;
+      this.surveyService.delete(this.survey)
+        .takeUntil(this.ngUnsubscribe)
+        .subscribe(
+          () => { },
+          error => {
+            console.log('error =', error);
```

Comment:
@Baelx Is this the best way to do this via console.log and an alert, maybe even consider using console.error, or have a logging function.